<h1 style="margin: auto; font-weight: bold; padding: 30px 30px 0px 30px; color:#000;" align="center">SCHOOL DROPOUT IN SOGAMOSO</h1>
<h3 style="width: 100%; text-align: center; float: left; font-size: 24px; color:#000;" align="center">| EDA Notebook |</h3>
<h4 style="width: 100%; text-align: center; float: left; font-size: 24px; color:#000;" align="center">Correlation One - DS4A</h4>
<p style="text-align:center;">In this notebook will be conducted an EDA in order to understand the data and how those data is related with the Schoold Dropout in Sogamoso, Colombia.</p>

## 1. Libraries

In [1]:
# General
import pandas as pd
import numpy as np

## Own specific functions 
from functions import *

## 2. Importing files

In [2]:
df_students_2019 = pd.read_csv("datasets/alumnos_s_oficial2019.csv", encoding="cp1252", sep=";")
df_students_2020 = pd.read_csv("datasets/alumnos_s_oficial2020.csv", encoding="cp1252", sep=";", low_memory=False)
df_students_2021 = pd.read_csv("datasets/alumnos_s_oficial2021.csv", encoding="cp1252", sep=";")
df_students_2022 = pd.read_csv("datasets/alumnos_s_oficial2022.csv", encoding="cp1252", sep=";")

### 2.1 Initial analysis

Let's make a initial analysis over the datasets

In [55]:
pd.set_option("display.max_columns", None) # To show all column in the dataset

for year in range(2019, 2022+1):
    print("\n>>>>>>>>>> df_students_" + str(year) + " <<<<<<<<<<")
    display(globals()["df_students_" + str(year)].head(3))
    df_analysis(globals()["df_students_" + str(year)],
                key_columns=["NUI"], analysis_type="complete")
    print("\n")
    
pd.reset_option("display.max_columns") # reset max of showing cols


>>>>>>>>>> df_students_2019 <<<<<<<<<<


ANO       ESTADO                                        INSTITUCION  \
0  2019  MATRICULADO          INSTITUCION EDUCATIVA COLEGIO DE SUGAMUXI   
1  2019  MATRICULADO  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...   
2  2019  MATRICULADO  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...   

       DANE                                               SEDE ZONA_SEDE  \
0  1,16E+11          INSTITUCION EDUCATIVA COLEGIO DE SUGAMUXI    URBANA   
1  1,16E+11  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...    URBANA   
2  1,16E+11  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...    URBANA   

  JORNADA  GRADO_COD GRUPO                 MODELO MOTIVO  \
0  MA¾ANA        9.0   902  EDUCACI…N TRADICIONAL    NaN   
1   ‚NICA        6.0   601  EDUCACI…N TRADICIONAL    NaN   
2   ‚NICA        6.0   601  EDUCACI…N TRADICIONAL    NaN   

                FECHAINI               NUI    ESTRATO     GENERO       SRPA  \
0  2019-04-23 15:50:51.0  DANABA1571788930  ESTRATO 1  MASCULINO  NO APLICA   
1  2019-02-21 09:34:38.0  ALEABE1931563776  ESTRATO 2  MASCULINO        NaN   
2  2019-04-01 09:26:21.0   YULABE571948373  ESTRATO 3  MASCULINO        NaN   

  DISCAPACIDAD PAIS_ORIGEN  
0    NO APLICA    COLOMBIA  
1    NO APLICA         NaN  
2    NO APLICA         NaN


Analysis header
--------------------------------------------------------------------------------
- Dataset shape:			 21710 rows and 18 columns
- Total of NaN values:			 44058
- Percentage of NaN:			 11.27 %
- Total of infinite values:		 0
- Total of empty columns:		 0
- Total of empty rows:			 0
- Total of full duplicates rows:	 0
- Unique indexes:			 True
- Memory usage:				 3.0+ MB

- The key(s):	 ['NUI'] is not present multiple times in the dataframe. 
		 It CAN be used as a primary key.


Detailed analysis
--------------------------------------------------------------------------------


name     type  records  unique  # NaN    % NaN       mean  \
0            ANO    int64    21710       1      0  0.00000 2019.00000   
1         ESTADO   object    21710      10      0  0.00000        NaN   
14        GENERO   object    21710       2      0  0.00000        NaN   
12           NUI   object    21710   21710      0  0.00000        NaN   
11      FECHAINI   object    21710   16656      0  0.00000        NaN   
13       ESTRATO   object    21708       8      2  0.01000        NaN   
16  DISCAPACIDAD   object    21705      15      5  0.02000        NaN   
7      GRADO_COD  float64    21701      19      9  0.04000    7.54062   
9         MODELO   object    21695       7     15  0.07000        NaN   
6        JORNADA   object    21695       5     15  0.07000        NaN   
5      ZONA_SEDE   object    21695       2     15  0.07000        NaN   
4           SEDE   object    21695      57     15  0.07000        NaN   
3           DANE   object    21695       2     15  0.07000        NaN   
2    INSTITUCION   object    21695      16     15  0.07000        NaN   
8          GRUPO   object    21667     178     43  0.20000        NaN   
15          SRPA   object    13610       3   8100 37.31000        NaN   
17   PAIS_ORIGEN   object     6653       4  15057 69.36000        NaN   
10        MOTIVO   object      958      10  20752 95.59000        NaN   

          min        25%        50%        75%        max     std  
0  2019.00000 2019.00000 2019.00000 2019.00000 2019.00000 0.00000  
1         NaN        NaN        NaN        NaN        NaN     NaN  
14        NaN        NaN        NaN        NaN        NaN     NaN  
12        NaN        NaN        NaN        NaN        NaN     NaN  
11        NaN        NaN        NaN        NaN        NaN     NaN  
13        NaN        NaN        NaN        NaN        NaN     NaN  
16        NaN        NaN        NaN        NaN        NaN     NaN  
7    -2.00000    3.00000    6.00000    9.00000   26.00000 6.47236  
9         NaN        NaN        NaN        NaN        NaN     NaN  
6         NaN        NaN        NaN        NaN        NaN     NaN  
5         NaN        NaN        NaN        NaN        NaN     NaN  
4         NaN        NaN        NaN        NaN        NaN     NaN  
3         NaN        NaN        NaN        NaN        NaN     NaN  
2         NaN        NaN        NaN        NaN        NaN     NaN  
8         NaN        NaN        NaN        NaN        NaN     NaN  
15        NaN        NaN        NaN        NaN        NaN     NaN  
17        NaN        NaN        NaN        NaN        NaN     NaN  
10        NaN        NaN        NaN        NaN        NaN     NaN




>>>>>>>>>> df_students_2020 <<<<<<<<<<


ANO       ESTADO                                        INSTITUCION  \
0  2020  MATRICULADO  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...   
1  2020  MATRICULADO  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...   
2  2020  MATRICULADO          INSTITUCION EDUCATIVA COLEGIO DE SUGAMUXI   

       DANE                                               SEDE ZONA_SEDE  \
0  1,16E+11  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...    URBANA   
1  1,16E+11  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...    URBANA   
2  1,16E+11          INSTITUCION EDUCATIVA COLEGIO DE SUGAMUXI    URBANA   

  JORNADA  GRADO_COD GRUPO                 MODELO MOTIVO  \
0   éNICA        7.0   701  EDUCACIàN TRADICIONAL    NaN   
1   éNICA        7.0   701  EDUCACIàN TRADICIONAL    NaN   
2  MA¥ANA       11.0  1104  EDUCACIàN TRADICIONAL    NaN   

                FECHAINI               NUI    ESTRATO     GENERO       SRPA  \
0  2020-01-31 08:00:23.0  ALEABE1931563776  ESTRATO 2  MASCULINO        NaN   
1  2020-01-31 08:00:59.0   YULABE571948373  ESTRATO 3  MASCULINO        NaN   
2  2019-12-18 07:45:41.0   ANYABE594991487  ESTRATO 2   FEMENINO  NO APLICA   

  DISCAPACIDAD PAIS_ORIGEN  
0    NO APLICA         NaN  
1    NO APLICA         NaN  
2    NO APLICA    COLOMBIA


Analysis header
--------------------------------------------------------------------------------
- Dataset shape:			 19994 rows and 18 columns
- Total of NaN values:			 30034
- Percentage of NaN:			 8.35 %
- Total of infinite values:		 0
- Total of empty columns:		 0
- Total of empty rows:			 0
- Total of full duplicates rows:	 0
- Unique indexes:			 True
- Memory usage:				 2.7+ MB

- The key(s):	 ['NUI'] is not present multiple times in the dataframe. 
		 It CAN be used as a primary key.


Detailed analysis
--------------------------------------------------------------------------------


name     type  records  unique  # NaN    % NaN       mean  \
0            ANO    int64    19994       1      0  0.00000 2020.00000   
1         ESTADO   object    19994       6      0  0.00000        NaN   
16  DISCAPACIDAD   object    19994      13      0  0.00000        NaN   
14        GENERO   object    19994       2      0  0.00000        NaN   
13       ESTRATO   object    19994       8      0  0.00000        NaN   
12           NUI   object    19994   19994      0  0.00000        NaN   
11      FECHAINI   object    19994   14270      0  0.00000        NaN   
7      GRADO_COD  float64    19993      19      1  0.01000    6.22813   
9         MODELO   object    19992       5      2  0.01000        NaN   
8          GRUPO   object    19992     136      2  0.01000        NaN   
6        JORNADA   object    19992       5      2  0.01000        NaN   
5      ZONA_SEDE   object    19992       2      2  0.01000        NaN   
4           SEDE   object    19992      56      2  0.01000        NaN   
3           DANE   object    19992       2      2  0.01000        NaN   
2    INSTITUCION   object    19992      16      2  0.01000        NaN   
15          SRPA   object    16654       3   3340 16.71000        NaN   
17   PAIS_ORIGEN   object    12635       8   7359 36.81000        NaN   
10        MOTIVO   object      674       4  19320 96.63000        NaN   

          min        25%        50%        75%        max     std  
0  2020.00000 2020.00000 2020.00000 2020.00000 2020.00000 0.00000  
1         NaN        NaN        NaN        NaN        NaN     NaN  
16        NaN        NaN        NaN        NaN        NaN     NaN  
14        NaN        NaN        NaN        NaN        NaN     NaN  
13        NaN        NaN        NaN        NaN        NaN     NaN  
12        NaN        NaN        NaN        NaN        NaN     NaN  
11        NaN        NaN        NaN        NaN        NaN     NaN  
7    -1.00000    3.00000    6.00000    9.00000   26.00000 4.63803  
9         NaN        NaN        NaN        NaN        NaN     NaN  
8         NaN        NaN        NaN        NaN        NaN     NaN  
6         NaN        NaN        NaN        NaN        NaN     NaN  
5         NaN        NaN        NaN        NaN        NaN     NaN  
4         NaN        NaN        NaN        NaN        NaN     NaN  
3         NaN        NaN        NaN        NaN        NaN     NaN  
2         NaN        NaN        NaN        NaN        NaN     NaN  
15        NaN        NaN        NaN        NaN        NaN     NaN  
17        NaN        NaN        NaN        NaN        NaN     NaN  
10        NaN        NaN        NaN        NaN        NaN     NaN




>>>>>>>>>> df_students_2021 <<<<<<<<<<


ANO       ESTADO                                        INSTITUCION  \
0  2021  MATRICULADO  INSTITUCION EDUCATIVA EMPRESARIAL Y AGROINDUST...   
1  2021  MATRICULADO  INSTITUCION EDUCATIVA EMPRESARIAL Y AGROINDUST...   
2  2021  MATRICULADO  INSTITUCION EDUCATIVA POLITECNICO ALVARO GONZA...   

       DANE                                               SEDE ZONA_SEDE  \
0  1,16E+11  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...    URBANA   
1  1,16E+11  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...    URBANA   
2  1,16E+11                            CONCENTRACION ELROSARIO    URBANA   

  JORNADA  GRADO_COD GRUPO                 MODELO MOTIVO         FECHAINI  \
0   éNICA        8.0   801  EDUCACIàN TRADICIONAL    NaN  3/09/2021 10:13   
1   éNICA        8.0   801  EDUCACIàN TRADICIONAL    NaN  3/09/2021 10:13   
2   TARDE        0.0     3  EDUCACIàN TRADICIONAL    NaN  11/02/2021 9:31   

                NUI    ESTRATO     GENERO       SRPA DISCAPACIDAD PAIS_ORIGEN  
0  ALEABE1931563776  ESTRATO 2  MASCULINO        NaN    NO APLICA         NaN  
1   YULABE571948373  ESTRATO 3  MASCULINO        NaN    NO APLICA         NaN  
2  YOAABE1310307342  ESTRATO 2  MASCULINO  NO APLICA    NO APLICA    COLOMBIA


Analysis header
--------------------------------------------------------------------------------
- Dataset shape:			 21562 rows and 18 columns
- Total of NaN values:			 27216
- Percentage of NaN:			 7.01 %
- Total of infinite values:		 0
- Total of empty columns:		 0
- Total of empty rows:			 0
- Total of full duplicates rows:	 0
- Unique indexes:			 True
- Memory usage:				 3.0+ MB

- The key(s):	 ['NUI'] is not present multiple times in the dataframe. 
		 It CAN be used as a primary key.


Detailed analysis
--------------------------------------------------------------------------------


name     type  records  unique  # NaN    % NaN       mean  \
0            ANO    int64    21562       1      0  0.00000 2021.00000   
1         ESTADO   object    21562       9      0  0.00000        NaN   
16  DISCAPACIDAD   object    21562      16      0  0.00000        NaN   
14        GENERO   object    21562       2      0  0.00000        NaN   
12           NUI   object    21562   21562      0  0.00000        NaN   
11      FECHAINI   object    21562    7976      0  0.00000        NaN   
13       ESTRATO   object    21561       7      1  0.00000        NaN   
7      GRADO_COD  float64    21560      20      2  0.01000    7.27055   
9         MODELO   object    21541       8     21  0.10000        NaN   
6        JORNADA   object    21541       5     21  0.10000        NaN   
5      ZONA_SEDE   object    21541       2     21  0.10000        NaN   
4           SEDE   object    21541      55     21  0.10000        NaN   
3           DANE   object    21541       2     21  0.10000        NaN   
2    INSTITUCION   object    21541      16     21  0.10000        NaN   
8          GRUPO   object    21540     187     22  0.10000        NaN   
15          SRPA   object    19401       3   2161 10.02000        NaN   
17   PAIS_ORIGEN   object    17320      12   4242 19.67000        NaN   
10        MOTIVO   object      900       6  20662 95.83000        NaN   

          min        25%        50%        75%        max     std  
0  2021.00000 2021.00000 2021.00000 2021.00000 2021.00000 0.00000  
1         NaN        NaN        NaN        NaN        NaN     NaN  
16        NaN        NaN        NaN        NaN        NaN     NaN  
14        NaN        NaN        NaN        NaN        NaN     NaN  
12        NaN        NaN        NaN        NaN        NaN     NaN  
11        NaN        NaN        NaN        NaN        NaN     NaN  
13        NaN        NaN        NaN        NaN        NaN     NaN  
7    -2.00000    3.00000    6.00000    9.00000   26.00000 6.14835  
9         NaN        NaN        NaN        NaN        NaN     NaN  
6         NaN        NaN        NaN        NaN        NaN     NaN  
5         NaN        NaN        NaN        NaN        NaN     NaN  
4         NaN        NaN        NaN        NaN        NaN     NaN  
3         NaN        NaN        NaN        NaN        NaN     NaN  
2         NaN        NaN        NaN        NaN        NaN     NaN  
8         NaN        NaN        NaN        NaN        NaN     NaN  
15        NaN        NaN        NaN        NaN        NaN     NaN  
17        NaN        NaN        NaN        NaN        NaN     NaN  
10        NaN        NaN        NaN        NaN        NaN     NaN




>>>>>>>>>> df_students_2022 <<<<<<<<<<


ANO       ESTADO                                        INSTITUCION  \
0  2022  MATRICULADO  INSTITUCION EDUCATIVA TECNICA INDUSTRIAL GUSTA...   
1  2022  MATRICULADO                    INSTITUCION EDUCATIVA MAGDALENA   
2  2022  MATRICULADO  INSTITUCION EDUCATIVA EMPRESARIAL Y AGROINDUST...   

       DANE                                               SEDE ZONA_SEDE  \
0  2,16E+11                                             RAMADA     RURAL   
1  1,16E+11                    INSTITUCION EDUCATIVA MAGDALENA    URBANA   
2  1,16E+11  INSTITUCION EDUCATIVA EMPRESARIALY AGROINDUSTR...    URBANA   

    JORNADA  GRADO_COD GRUPO                                        MODELO  \
0     TARDE        0.0     1                         EDUCACIàN TRADICIONAL   
1  NOCTURNA       22.0   201  PROGRAMA PARA JàVENES EN EXTRAEDAD Y ADULTOS   
2     éNICA        9.0   901                         EDUCACIàN TRADICIONAL   

  MOTIVO          FECHAINI               NUI    ESTRATO     GENERO       SRPA  \
0    NaN    8/02/2022 8:29   OBEABA297269100  ESTRATO 2   FEMENINO  NO APLICA   
1    NaN   1/03/2022 19:29  EBEABA1354365140  ESTRATO 3  MASCULINO  NO APLICA   
2    NaN  19/01/2022 15:51  ALEABE1931563776  ESTRATO 2  MASCULINO        NaN   

  DISCAPACIDAD PAIS_ORIGEN  
0    NO APLICA   VENEZUELA  
1    NO APLICA    COLOMBIA  
2    NO APLICA         NaN


Analysis header
--------------------------------------------------------------------------------
- Dataset shape:			 21111 rows and 18 columns
- Total of NaN values:			 25613
- Percentage of NaN:			 6.74 %
- Total of infinite values:		 0
- Total of empty columns:		 0
- Total of empty rows:			 0
- Total of full duplicates rows:	 0
- Unique indexes:			 True
- Memory usage:				 2.9+ MB

- The key(s):	 ['NUI'] is not present multiple times in the dataframe. 
		 It CAN be used as a primary key.


Detailed analysis
--------------------------------------------------------------------------------


name     type  records  unique  # NaN    % NaN       mean  \
0            ANO    int64    21111       1      0  0.00000 2022.00000   
16  DISCAPACIDAD   object    21111      15      0  0.00000        NaN   
14        GENERO   object    21111       2      0  0.00000        NaN   
12           NUI   object    21111   21111      0  0.00000        NaN   
11      FECHAINI   object    21111    7746      0  0.00000        NaN   
1         ESTADO   object    21111       8      0  0.00000        NaN   
7      GRADO_COD  float64    21109      19      2  0.01000    7.05017   
9         MODELO   object    21101       7     10  0.05000        NaN   
6        JORNADA   object    21101       5     10  0.05000        NaN   
5      ZONA_SEDE   object    21101       2     10  0.05000        NaN   
4           SEDE   object    21101      54     10  0.05000        NaN   
3           DANE   object    21101       2     10  0.05000        NaN   
2    INSTITUCION   object    21101      16     10  0.05000        NaN   
13       ESTRATO   object    21100       8     11  0.05000        NaN   
8          GRUPO   object    21077     178     34  0.16000        NaN   
15          SRPA   object    19403       3   1708  8.09000        NaN   
17   PAIS_ORIGEN   object    17700      15   3411 16.16000        NaN   
10        MOTIVO   object      724       4  20387 96.57000        NaN   

          min        25%        50%        75%        max     std  
0  2022.00000 2022.00000 2022.00000 2022.00000 2022.00000 0.00000  
16        NaN        NaN        NaN        NaN        NaN     NaN  
14        NaN        NaN        NaN        NaN        NaN     NaN  
12        NaN        NaN        NaN        NaN        NaN     NaN  
11        NaN        NaN        NaN        NaN        NaN     NaN  
1         NaN        NaN        NaN        NaN        NaN     NaN  
7    -1.00000    3.00000    6.00000    9.00000   26.00000 5.94476  
9         NaN        NaN        NaN        NaN        NaN     NaN  
6         NaN        NaN        NaN        NaN        NaN     NaN  
5         NaN        NaN        NaN        NaN        NaN     NaN  
4         NaN        NaN        NaN        NaN        NaN     NaN  
3         NaN        NaN        NaN        NaN        NaN     NaN  
2         NaN        NaN        NaN        NaN        NaN     NaN  
13        NaN        NaN        NaN        NaN        NaN     NaN  
8         NaN        NaN        NaN        NaN        NaN     NaN  
15        NaN        NaN        NaN        NaN        NaN     NaN  
17        NaN        NaN        NaN        NaN        NaN     NaN  
10        NaN        NaN        NaN        NaN        NaN     NaN

<div class="alert alert-block alert-info">
    <p><b>Observations</b></p>
    <p>Below, some points observed:</p>
    <p>
        <ul style="list-style-type: square;">
            <li>It appears, the number of children differs in each list. It is necessary to validate how much students are into each dataset.</li>
            <li>There are some variables that do not match between datasetsets</li>
            <li>There is a primary key into all dataset ['NUI']  that is not present multiple times in each dataframe</li>
            <li>There are some columns (ETC, SECTOR, CALENDARIO) that only have one value so, we are going to delete them</li>
            <li>Most of the column in each dataset are object type</li>
        </ul>
    </p>
    <p>Now, a little detail for each of the datasets</p>
    <p>
        <ul style="list-style-type: square;">
            <li><b>df_alumnos_2019</b></li>
            <ul style="list-style-type: disc;">
               <li>Dataset shape: 21710 rows and 31 columns</li>
               <li>Missing values: 98130 / 14.58%</li>
               <li>Key(s) column(s): Nui can be used as a primary key</li>
            </ul>
        </ul>
        <ul style="list-style-type: square;">
            <li><b>df_alumnos_2020</b></li>
            <ul style="list-style-type: disc;">
               <li>Dataset shape: 19994 rows and 32 columns</li>
               <li>Missing values: 80991 / 12.66 %</li>
               <li>Key(s) column(s): Nui can be used as a primary key</li>
            </ul>
        </ul>
        <ul style="list-style-type: square;">
            <li><b>df_alumnos_2021</b></li>
            <ul style="list-style-type: disc;">
               <li>Dataset shape: 21562 rows and 26 columns</li>
               <li>Missing values: 70984 / 12.66 %</li>
               <li>Key(s) column(s): Nui can be used as a primary key</li>
            </ul>
        </ul>
        <ul style="list-style-type: square;">
            <li><b>df_alumnos_2021</b></li>
            <ul style="list-style-type: disc;">
               <li>Dataset shape: 21111 rows and 34 columns</li>
               <li>Missing values: 117357 / 16.35 %</li>
               <li>Key(s) column(s): Nui can be used as a primary key</li>
            </ul>
        </ul>
    </p>
</div>

Next, the variables that do not match between datasetsets

![Variables that do not match between datasets](images/variables_that_do_not_match.png)

## 3. Cleaning

### 3.1 Initial cleaning

Deleting columns and rows with 100% of missing-values. At the same time, let's delete the duplicated rows

In [18]:
for year in range(2019, 2022+1):
    globals()["df_students_" + str(year)] = globals()["df_students_" + str(year)].\
                                                dropna(axis="columns", how="all").dropna(axis="rows", how="all")

### 3.2 Removing unnecessary columns

Now, let's remove some unnecessary columns identified by Sogamoso's Municipality and also, thanks to initial analysis. 

In [14]:
COLUMNS_TO_DELETE_THAT_DO_NOT_MATCH = [
    "CODIGO_DANE_SEDE", "CONSECUTIVO", "SISBEN IV", "SISBEN_TRES",
    "PER_ID", "TIPO DE SANGRE", "MATRICULACONTRATADA", "FUENTE_RECURSOS",
    "INTERNADO", "NUM_CONTRATO", "APOYO_ACADEMICO_ESPECIAL"
]

OTHER_COLUMNS_TO_DELETE = [
    "JERARQUIA", "ETC", "SECTOR", "CALENDARIO", "Unnamed: 33", "Unnamed: 25"
]

COLUMNS_TO_DELETE = COLUMNS_TO_DELETE_THAT_DO_NOT_MATCH + OTHER_COLUMNS_TO_DELETE

for col in COLUMNS_TO_DELETE:
    for year in range(2019, 2022+1):
        if col in globals()["df_students_" + str(year)]:
            del globals()["df_students_" + str(year)][col]

<div class="alert alert-block alert-warning">
    <p><b>Validate with Sogamoso's Municipality</b></p>
    <p>column SRPA</p>
</div>

In [24]:
df_students_2019["SRPA"].unique()

array(['NO APLICA', nan, 'PRIVADO DE LA LIBERTAD',
       'NO PRIVADO DE LA LIBERTAD'], dtype=object)

### 3.3 Checking columns names

In [38]:
for year in range(2019, 2022+1):
    print("df_students_" + str(year))
    print(list(globals()["df_students_" + str(year)].columns),"\n")

df_students_2019
['ANO', 'ESTADO', 'INSTITUCION', 'DANE', 'SEDE', 'ZONA_SEDE', 'JORNADA', 'GRADO_COD', 'GRUPO', 'MODELO', 'MOTIVO', 'FECHAINI', 'NUI', 'ESTRATO', 'GENERO', 'SRPA', 'DISCAPACIDAD', 'PAIS_ORIGEN'] 

df_students_2020
['ANO', 'ESTADO', 'INSTITUCION', 'DANE', 'SEDE', 'ZONA_SEDE', 'JORNADA', 'GRADO_COD', 'GRUPO', 'MODELO', 'MOTIVO', 'FECHAINI', 'NUI', 'ESTRATO', 'GENERO', 'SRPA', 'DISCAPACIDAD', 'PAIS_ORIGEN'] 

df_students_2021
['ANO', 'ESTADO', 'INSTITUCION', 'DANE', 'SEDE', 'ZONA_SEDE', 'JORNADA', 'GRADO_COD', 'GRUPO', 'MODELO', 'MOTIVO', 'FECHAINI', 'NUI', 'ESTRATO', 'GENERO', 'SRPA', 'DISCAPACIDAD', 'PAIS_ORIGEN'] 

df_students_2022
['ANO', 'ESTADO', 'INSTITUCION', 'DANE', 'SEDE', 'ZONA_SEDE', 'JORNADA', 'GRADO_COD', 'GRUPO', 'MODELO', 'MOTIVO', 'FECHAINI', 'NUI', 'ESTRATO', 'GENERO', 'SRPA', 'DISCAPACIDAD', 'PAIS_ORIGEN'] 



<div class="alert alert-block alert-info">
    <p><b>Observations</b></p>
    <p>The columns name are equals into all datasets</p>
</div>

<div class="alert alert-block alert-danger">
    <p><b>Flag!</b></p>
</div>

In [59]:
for year in range(2019, 2022+1):
    print("\n>>>>>>>>>> df_students_" + str(year) + " <<<<<<<<<<")
    df_analysis(globals()["df_students_" + str(year)],
                analysis_type="header")


>>>>>>>>>> df_students_2019 <<<<<<<<<<

Analysis header
--------------------------------------------------------------------------------
- Dataset shape:			 21710 rows and 18 columns
- Total of NaN values:			 44058
- Percentage of NaN:			 11.27 %
- Total of infinite values:		 0
- Total of empty columns:		 0
- Total of empty rows:			 0
- Total of full duplicates rows:	 0
- Unique indexes:			 True
- Memory usage:				 3.0+ MB

>>>>>>>>>> df_students_2020 <<<<<<<<<<

Analysis header
--------------------------------------------------------------------------------
- Dataset shape:			 19994 rows and 18 columns
- Total of NaN values:			 30034
- Percentage of NaN:			 8.35 %
- Total of infinite values:		 0
- Total of empty columns:		 0
- Total of empty rows:			 0
- Total of full duplicates rows:	 0
- Unique indexes:			 True
- Memory usage:				 2.7+ MB

>>>>>>>>>> df_students_2021 <<<<<<<<<<

Analysis header
--------------------------------------------------------------------------------
- Datas